br/pkg/checkpoint/checkpoint.go

```diff
@@ -463,24 +457,23 @@ func (r *CheckpointRunner[K, V]) startCheckpointMainLoop(
 					r.meta[msg.GroupKey] = groups
 				}
 				groups.Group = append(groups.Group, msg.Group...)
-			case msg, ok := <-r.checksumCh:
-				if !ok {
-					log.Info("stop checkpoint runner")
-					if err := r.flushMeta(ctx, errCh); err != nil {
-						r.sendError(err)
-					}
-					if err := r.flushChecksum(ctx, errCh); err != nil {
-						r.sendError(err)
-					}
-					// close the channel to flush worker
-					// and wait it to consumes all the metas
-					close(r.metaCh)
-					close(r.checksumMetaCh)
-					close(r.lockCh)
-					wg.Wait()
-					return
-				}
+			case msg := <-r.checksumCh:
 				r.checksum.Items = append(r.checksum.Items, msg)
+			case <-r.doneCh:
```

Comment:
The snapshot restore uses the pipeline. So if the restore failed, then call `WaitForFinish` to close the channel, the checksum goroutine would send some data to the channel, which leads to panic.